ui/app/components/ttl-picker2.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @module TtlPicker2
+ * TtlPicker2 components are used to enable and select TTL
+ *
+ * @example
+ * ```js
+ * <TtlPicker2 @requiredParam={requiredParam} @optionalParam={optionalParam} @param1={{param1}}/>
+ * ```
+ * @param {function} onChange - This function will be passed a TTL object, which includes enabled{bool}, seconds{number}, timeString{string}.
+ * @param {string} [label='Time to live (TTL)'] - Label is the main label that lives next to the toggle.
+ * @param {string} [helperTextDisabled='Allow tokens to be used indefinitely'] - This helper text is shown under the label when the toggle is switched off
+ * @param {string} [helperTextEnabled='Disable the use of the token after'] - This helper text is shown under the label when the toggle is switched on
+ * @param {number} [time=30] - This is the time (in the default units) which will be adjustable by the user of the form
+ * @param {string} [unit='s'] - This is the unit key which will show by default on the form. Can be one of `s` (seconds), `m` (minutes), `h` (hours), `d` (days)
+ */
+
+import Ember from 'ember';
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { task, timeout } from 'ember-concurrency';
+
+const secondsMap = {
+  s: 1,
+  m: 60,
+  h: 3600,
+  d: 86400,
+};
+const convertToSeconds = (time, unit) => {
+  return time * secondsMap[unit];
+};
+const convertFromSeconds = (seconds, unit) => {
+  return seconds / secondsMap[unit];
+};
+
+export default Component.extend({
+  enableTTL: false,
+  label: 'Time to live (TTL)',
+  helperTextDisabled: 'Allow tokens to be used indefinitely',
```

Comment:
Because this text is within a component, I'm guessing it can't be changed if someone were to use this component elsewhere?  If that's the case, can we pass this text in as props so that it can be changed depending on how it's being used?  Same with `helperTextEnabled`.

Reply:
This might be me misunderstanding how the params work. I was intending this to be the default text, which can be overwritten (as shown in the nested story), and I had this on default because in general, I expect us to standardize on the language we use for the TTL helper text.

Reply:
@chelshaw yep! your understanding of params is correct, and your use of default text here is  👍

Reply:
Ah got it.  Thanks!